### PR TITLE
genpolicy: add topologySpreadConstraints support

### PR DESCRIPTION
**Pull request by Redent0r**
Allow genpolicy to process Pod YAML files including topologySpreadConstraints.

### CODE REVIEW

File: tests/integration/kubernetes/runtimeclass_workloads/k8s-policy-pod.yaml [hunk 1]
```diff
@@ -27,3 +27,7 @@ spec:
       securityContext:
         seccompProfile:
           type: RuntimeDefault
+  topologySpreadConstraints:
+        - maxSkew: 2
```
Comment (wainersm) [inline tests/integration/kubernetes/runtimeclass_workloads/k8s-policy-pod.yaml:31]: could you please keep it 2-spaces indentation?

Comment (wainersm) [reply tests/integration/kubernetes/runtimeclass_workloads/k8s-policy-pod.yaml:31]: That's the only comment I have about this PR. I check the pod spec in https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/ and this is handling all the fields (optional and required).

Comment (Redent0r) [reply tests/integration/kubernetes/runtimeclass_workloads/k8s-policy-pod.yaml]: Updated. Thanks @wainersm !

**Review by wainersm — APPROVED** [review]
Thanks @Redent0r !

**Review by stevenhorsman — APPROVED** [review]
I don't necessarily understand the option we are enabling policy for, but the code looks okay and the tests pass, so I think it's ok to go in.

**Pull request merged**